qiskit_metal/analyses/quantization/lom_core_analysis.py

```diff
@@ -0,0 +1,1267 @@
+    """
+    mat_ = np.delete(mat, idx, 0)
+    mat_ = np.delete(mat_, idx, 1)
+    return mat_
+
+
+def _extract_matrix_from_transform(transform: BasisTransform,
+                                   junctions: Mapping[Tuple[str, str], str],
+                                   index: pd.Index) -> np.ndarray:
+    """
+    calculate the S_n^{-1} matrix that corresponds to a given transform
+    https://arxiv.org/pdf/2103.10344.pdf
+    """
+    junctions = dict(zip(junctions.values(), junctions.keys()))
+
+    dim = len(transform.orig_node_basis)
+    m = np.zeros((dim, dim))
+    for ii, (n_old, n_new) in enumerate(
+            zip(transform.orig_node_basis, transform.node_jj_basis)):
+        if n_old == n_new:
+            m[ii, ii] = 1
+        else:
+            n1, n2 = junctions[n_new]
+            c1, c2 = index.get_indexer([n1, n2])
+            m[ii, c1] = 1
+            m[ii, c2] = -1
+    return m
+
+
+def _transform_to_junction_flux_basis(orig_nodes,
+                                      junctions: Mapping[Tuple[str, str], str],
+                                      choose_least_num_neg: bool = True):
+    """
+    junctions = {('n1', 'n2'):'j1', ('n3', 'n4'):'j2'}
+    junction flux is defined such that Phi_{j1} = Phi_{1} - Phi_{2}, for {'j1': ('n1', 'n2')}, where we refer to n1 as
+    the "positive" node and n2 as the "negative" node
+
+    Param:
+        choose_least_num_neg: if set true, choose the transform with least number of -1s in the transform matrix.
+            this means choosing a basis which has the LARGEST number of "positive" junction nodes, i.e.,
+            the LEAST number of "negative" junction nodes; if set false, return all possible transforms
+    """
+    junctions = dict(zip(junctions.values(), junctions.keys()))
+
+    j_list = list(junctions.keys())
+
+    # each junction flux can potentially replace one of two orignal nodes in the new flux basis.
+    # for N junctions, an array of N nodes can represent such mapping where the ith node in the array
+    # is the original node that is replaced by the ith junction; And an array of these arrays can
+    # represent all possible mappings
+
+    all_combos = _product_no_duplicate(*tuple(junctions.values()))
+    all_maps = [dict(zip(x, j_list)) for x in all_combos]
+
+    _transforms = []
+    for _map in all_maps:
+        new_basis = []
+        num_negative_nodes = 0
+        for n in orig_nodes:
+            if n not in _map:
+                new_basis.append(n)
+            else:
+                new_basis.append(_map[n])
+                if junctions[_map[n]][1] == n:  # "negative" node
+                    num_negative_nodes += 1
+
+        _transforms.append(
+            BasisTransform(orig_nodes, new_basis, num_negative_nodes))
+
+    if choose_least_num_neg:
+        _transforms.sort(key=lambda t: t.num_negative_nodes)
+        return _transforms[0]
+
+    return _transforms
+
+
+class LabeledNdarray(np.ndarray):
```

Comment:
This is pretty interesting. i can't recall if you presented on this implementation in the meeting. Maybe can go over this next time, motivations, and implementation